packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -345,8 +345,11 @@ async function initializePackageEnvironment(locator: Locator, {project, binFolde
 
     const env = await makeScriptEnv({project, binFolder, lifecycleScript});
 
-    for (const [binaryName, [, binaryPath]] of await getPackageAccessibleBinaries(locator, {project}))
-      await makePathWrapper(binFolder, toFilename(binaryName), process.execPath, [binaryPath]);
+    await Promise.all(
+      Array.from(await getPackageAccessibleBinaries(locator, {project}), ([binaryName, [, binaryPath]]) =>
+        makePathWrapper(binFolder, toFilename(binaryName), process.execPath, [binaryPath])
```

Comment:
@arcanis Is it a bug that this doesn't use `env.BERRY_BIN_FOLDER`? :thinking:

Reply:
It was intended in the sense that `BERRY_BIN_FOLDER` is suppose to be the same as `binFolder`, except that it's not a portable path - so better use the original variable when you need a portable one.

Then the `setupScriptEnvironment` hook got added and now script have an opportunity to moify `BERRY_BIN_FOLDER`, so this isn't necessarily true anymore 🤔